### PR TITLE
Report WorkProcessorOperator statistics

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by sopel39** [comment]
With `work_processor_pipelines`:
```
presto> explain analyze select * from hive.tpch_sf10_orc.lineitem order by partkey limit 10;
                                                                                                                                                                 Query Plan
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [SINGLE]
     CPU: 13.04ms, Scheduled: 20.31ms, Input: 480 rows (71.47kB); per task: avg.: 480.00 std.dev.: 0.00, Output: 10 rows (1.54kB)
     Output layout: [orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, returnflag, linestatus, shipdate, commitdate, receiptdate, shipinstruct, shipmode, comment]
     Output partitioning: SINGLE []
     Grouped Execution: false
     - TopN[10 by (partkey ASC_NULLS_LAST)]
             Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:varch
             CPU: 2.00ms (0.00%), Scheduled: 2.00ms (0.01%), Output: 10 rows (1.54kB)
             Input avg.: 480.00 rows, Input std.dev.: 0.00%
         - LocalExchange[SINGLE] ()
                 Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:v
                 Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
                 CPU: 3.00ms (0.00%), Scheduled: 5.00ms (0.03%), Output: 480 rows (71.47kB)
                 Input avg.: 30.00 rows, Input std.dev.: 272.34%
             - RemoteSource[2]
                     Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestat
                     CPU: 2.00ms (0.00%), Scheduled: 4.00ms (0.02%), Output: 480 rows (71.47kB)
                     Input avg.: 30.00 rows, Input std.dev.: 272.34%

 Fragment 2 [SOURCE]
     CPU: 19.90s, Scheduled: 3.99m, Input: 59986052 rows (803.03MB); per task: avg.: 59986052.00 std.dev.: 0.00, Output: 480 rows (71.47kB)
     Output layout: [orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, returnflag, linestatus, shipdate, commitdate, receiptdate, shipinstruct, shipmode, comment]
     Output partitioning: SINGLE []
     Grouped Execution: false
     - TopNPartial[10 by (partkey ASC_NULLS_LAST)]
             Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:varch
             CPU: 18.01s (4.08%), Scheduled: 2.93m (883.29%), Output: 480 rows (71.47kB)
             Input avg.: 1153577.92 rows, Input std.dev.: 30.50%
         - TableScan[hive:tpch_sf10_orc:lineitem, grouped = false]
                 Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:v
                 Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}
                 CPU: 1.89s (0.43%), Scheduled: 4.42m (1332.40%), Output: 59986052 rows (803.03MB)
                 Input avg.: 1153577.92 rows, Input std.dev.: 30.50%
                 returnflag := returnflag:varchar(1):8:REGULAR
                 linenumber := linenumber:int:3:REGULAR
                 quantity := quantity:double:4:REGULAR
                 orderkey := orderkey:bigint:0:REGULAR
                 shipmode := shipmode:varchar(10):14:REGULAR
                 discount := discount:double:6:REGULAR
                 tax := tax:double:7:REGULAR
                 suppkey := suppkey:bigint:2:REGULAR
                 partkey := partkey:bigint:1:REGULAR
                 shipinstruct := shipinstruct:varchar(25):13:REGULAR
                 linestatus := linestatus:varchar(1):9:REGULAR
                 extendedprice := extendedprice:double:5:REGULAR

Query 20190518_203533_00012_2e295, FINISHED, 1 node
Splits: 86 total, 86 done (100,00%)
0:17 [60M rows, 1,58GB] [3,59M rows/s, 96,7MB/s]
```

Without:
```
presto> explain analyze select * from hive.tpch_sf10_orc.lineitem order by partkey limit 10;
                                                                                                                                                                 Query Plan
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [SINGLE]
     CPU: 16.72ms, Scheduled: 41.81ms, Input: 480 rows (71.47kB); per task: avg.: 480.00 std.dev.: 0.00, Output: 10 rows (1.52kB)
     Output layout: [orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, returnflag, linestatus, shipdate, commitdate, receiptdate, shipinstruct, shipmode, comment]
     Output partitioning: SINGLE []
     Grouped Execution: false
     - TopN[10 by (partkey ASC_NULLS_LAST)]
             Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:varch
             CPU: 2.00ms (0.00%), Scheduled: 5.00ms (0.01%), Output: 10 rows (1.52kB)
             Input avg.: 480.00 rows, Input std.dev.: 0.00%
         - LocalExchange[SINGLE] ()
                 Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:v
                 Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
                 CPU: 3.00ms (0.00%), Scheduled: 8.00ms (0.02%), Output: 480 rows (71.47kB)
                 Input avg.: 30.00 rows, Input std.dev.: 281.12%
             - RemoteSource[2]
                     Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestat
                     CPU: 3.00ms (0.00%), Scheduled: 5.00ms (0.01%), Output: 480 rows (71.47kB)
                     Input avg.: 30.00 rows, Input std.dev.: 281.12%

 Fragment 2 [SOURCE]
     CPU: 39.84s, Scheduled: 4.95m, Input: 59986052 rows (7.30GB); per task: avg.: 59986052.00 std.dev.: 0.00, Output: 480 rows (71.47kB)
     Output layout: [orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, returnflag, linestatus, shipdate, commitdate, receiptdate, shipinstruct, shipmode, comment]
     Output partitioning: SINGLE []
     Grouped Execution: false
     - TopNPartial[10 by (partkey ASC_NULLS_LAST)]
             Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:varch
             CPU: 4.66s (0.89%), Scheduled: 7.98s (20.01%), Output: 480 rows (71.47kB)
             Input avg.: 1153577.92 rows, Input std.dev.: 30.50%
         - TableScan[hive:tpch_sf10_orc:lineitem, grouped = false]
                 Layout: [orderkey:bigint, partkey:bigint, suppkey:bigint, linenumber:integer, quantity:double, extendedprice:double, discount:double, tax:double, returnflag:varchar(1), linestatus:v
                 Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}
                 CPU: 35.18s (6.75%), Scheduled: 8.55m (1287.39%), Output: 59986052 rows (7.30GB)
                 Input avg.: 1153577.92 rows, Input std.dev.: 30.50%
                 returnflag := returnflag:varchar(1):8:REGULAR
                 linenumber := linenumber:int:3:REGULAR
                 quantity := quantity:double:4:REGULAR
                 orderkey := orderkey:bigint:0:REGULAR
                 shipmode := shipmode:varchar(10):14:REGULAR
                 discount := discount:double:6:REGULAR
                 tax := tax:double:7:REGULAR
                 suppkey := suppkey:bigint:2:REGULAR
                 partkey := partkey:bigint:1:REGULAR
                 shipinstruct := shipinstruct:varchar(25):13:REGULAR
                 linestatus := linestatus:varchar(1):9:REGULAR
                 extendedprice := extendedprice:double:5:REGULAR
                 comment := comment:varchar(44):15:REGULAR
                 receiptdate := receiptdate:date:12:REGULAR
                 commitdate := commitdate:date:11:REGULAR
                 shipdate := shipdate:date:10:REGULAR


(1 row)

Query 20190518_203449_00010_2e295, FINISHED, 1 node
Splits: 86 total, 86 done (100,00%)
0:20 [60M rows, 1,61GB] [3,05M rows/s, 83,8MB/s]
```

File: presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java [hunk 1]
```diff
@@ -200,6 +220,11 @@ else if (state.getType() == BLOCKED && blockedFuture != state.getBlocked()) {
         }
     }
 
+    private long deltaAndSet(AtomicLong currentValue, long newValue)
```
Comment (dain) [inline presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java]: can this be static?

File: presto-main/src/main/java/io/prestosql/operator/WorkProcessorOperatorFactory.java [hunk 2]
```diff
@@ -16,11 +16,16 @@
...
 import io.prestosql.spi.Page;
+import io.prestosql.sql.planner.plan.PlanNodeId;
 
 public interface WorkProcessorOperatorFactory
 {
     int getOperatorId();
 
+    PlanNodeId getPlanNodeId();
+
+    String getOperatorType();
```
[the remark this answers is not in the document]
Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/operator/WorkProcessorOperatorFactory.java]: Factory class is not the same as operator class.

File: presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java [hunk 3]
```diff
@@ -406,14 +426,98 @@ private static Throwable handleOperatorCloseError(Throwable inFlightException, T
...
+        {
+            ListenableFuture<?> blocked = delegate.setBytes(bytes);
+            allocationListener.run();
+            return blocked;
+        }
+
+        @Override
+        public boolean trySetBytes(long bytes)
+        {
+            return delegate.trySetBytes(bytes);
```
[the remark this answers is not in the document]
Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java]: You have discovered another mem leak in `OperatorContext` mem tracking

**Pull request merged**